### PR TITLE
Add flags in CapsuleX64.inf to avoid alignment changes

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Set necessary build option flags to avoid the alignment change introduced with the Memory Attributes Table.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested with CI

## Integration Instructions

N/A

### CODE REVIEW

File: MdePkg/Library/VsIntrinsicLib/IA32/llmul.asm [hunk 1]
```diff
@@ -0,0 +1,98 @@
+;***
+;llmul.asm - long multiply routine
+;
+;       Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (makubacki) [inline MdePkg/Library/VsIntrinsicLib/IA32/llmul.asm]:
```suggestion
; Copyright (c) Microsoft Corporation.
```

File: MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf [hunk 2]
```diff
@@ -50,3 +50,11 @@
 
 [UserExtensions.TianoCore."ExtraFiles"]
   CapsuleX64Extra.uni
+
+## MS_CHANGE_211478
```
Comment (makubacki) [inline MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf]: Can you please change these to `MU_CHANGE`?

File: MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf [hunk 3]
```diff
@@ -50,3 +50,11 @@
 
 [UserExtensions.TianoCore."ExtraFiles"]
   CapsuleX64Extra.uni
+
+## MS_CHANGE_211478
+## MSChange [BEGIN] - This is necessary to avoid the alignment change introduced with the Memory Attributes Table
```
Comment (makubacki) [inline MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf]: This change seems it is for a different problem than the current PR scope which is "Fixing compilation warnings/errors from GCC on Linux in MU Code".

Comment (makubacki) [reply MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf]: In general, the PR title and body don't seem to match. I'll come back later.

File: MdePkg/Library/VsIntrinsicLib/IA32/llmul.asm (hunk 1 above)
Comment (kenlautner) [reply MdePkg/Library/VsIntrinsicLib/IA32/llmul.asm]: Fixed.

File: MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf (hunk 2 above)
Comment (kenlautner) [reply MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf]: Updated.

File: MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf (hunk 3 above)
Comment (os-d) [reply MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf]: We can drop this, it got updated by adding NX support: https://github.com/microsoft/mu_basecore/commit/ac74e7e44de48181e2ce33062c48d794a3d53df8. I am carrying a version of that commit here: https://github.com/microsoft/mu_basecore/pull/912/commits/4c6c82e3b8c089fa88c56de566a0d6e947f36e18.

I'm not sure the history here, it doesn't really feel related to memory protections, but I don't know why it is separate. I'm also fine to pull it out of the memory protections PR and put it here, just taking the updated lines.

Comment (kenlautner) [reply MdeModulePkg/Universal/CapsulePei/CapsuleX64.inf]: I slimmed down the PR considerably and updated the PR title and description to match the changes.

**Pull request closed without merging**